File: unidade01/conceitos/relacional/paciente.js
import { Pessoa } from "./pessoa.js";

export class Paciente extends Pessoa {
  #doenca;
  constructor(nome, idade, doenca) {
    super(nome, idade);
    this.#doenca = doenca;
  }

  solicitar_procedimento(medico) {
    console.log(
      `O paciente ${this.getNome()} solicitou atendimento ao ${medico.getNome()}.`,
    );
    medico.realizar_procedimento();
  }
}

File: estudo-aoo/src/relacional/Paciente.java
package relacional;

public class Paciente extends Pessoa {
    private String doenca;

    // Constructor
    public Paciente(String nome, int idade, String doenca){
        super(nome, idade);
        this.doenca = doenca;
    }

    // Getters e Setters
    public String getNome(){
        return nome;
    }

    public int getIdade() {
        return super.getIdade();
    }

    public void setNome(String nome){
        this.nome = nome;
    }

    public void setIdade(int idade){
        this.idade = idade;
    }

    public void solicitarProcedimento(Medico medico) {
        System.out.println("O paciente " + this.nome + " solicitou um procedimento ao médico " + medico.getNome());
        medico.realizarProcedimento();
    }
}
